pkg/client/devops/jenkins/pure_request.go

```diff
@@ -48,12 +48,26 @@ func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *dev
 	client := &http.Client{Timeout: 30 * time.Second}
 
 	header := httpParameters.Header
+	if header == nil {
+		header = http.Header{}
+	}
 
 	if j.Requester != nil {
 		auth := j.Requester.BasicAuth
 
 		creds := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", auth.Username, auth.Password)))
 		header.Set("Authorization", fmt.Sprintf("Basic %s", creds))
+
+		// we only set crumb to request header for POST method
+		// See also https://github.com/jenkinsci/jenkins/blob/ca646b3742e6921d332078d0b287fd359970d7e9/core/src/main/java/hudson/security/csrf/CrumbFilter.java#L126
+		if httpParameters.Method == http.MethodPost {
+			if err := j.Requester.SetCrumbForConsumer(func(crumbRequestField, crumb string) {
+				header.Set(crumbRequestField, crumb)
+			}); err != nil {
+				klog.Errorf("unable to set crumb to http parameters, err: %v", err)
```

Comment:
`http` -> `HTTP`

Crumb should be in the HTTP header instead of parameters.

Reply:
Alright.